implement CLICOLOR_FORCE environment variable

Mic92 opened the pull request:
This allows to force colors but not disable other features such as the progress bar. This is for instance useful in CI environments.

Adapted from
https://git.lix.systems/lix-project/lix/commit/378ec5fb0611e314511a7afc806664205846fc2e and others.

Tested in https://buildbot.thalheim.io/#/builders/103/builds/1490

<!--

IMPORTANT

Nix is a non-trivial project, so for your contribution to be successful,
it really is important to follow the contributing guidelines:

https://github.com/NixOS/nix/blob/master/CONTRIBUTING.md

Even if you've contributed to open source before, take a moment to read it,
so you understand the process and the expectations.

- what information to include in commit messages
- proper attribution
- volunteering contributions effectively
- how to get help and our review process.

-->

## Motivation

<!-- Briefly explain what the change is about and why it is desirable. -->

## Context

<!-- Provide context. Reference open issues if available. -->

<!-- Non-trivial change: Briefly outline the implementation strategy. -->

<!-- Invasive change: Discuss alternative designs or approaches you considered. -->

<!-- Large change: Provide instructions to reviewers how to read the diff. -->

---

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

Mic92 commented:
cc @xokdvium

Mic92 commented:
This still spawns a progressbar

cole-h commented on src/libutil/terminal.cc:
```diff
@@ -61,14 +61,28 @@ inline std::pair<int, size_t> charWidthUTF8Helper(std::string_view s)
+    // Implements the behaviour described by https://bixense.com/clicolors/
+    // As well as https://force-color.org/ for compatibility, since it fits in the same shape.
+    // NO_COLOR CLICOLOR CLICOLOR_FORCE Colours?
+    // set      x        x              No
+    // unset    x        set            Yes
+    // unset    x        unset          If attached to a terminal
+    //                                  [we choose the "modern" approach of colour-by-default]
```
I'm assuming the `x` means "any value"...? Could you just write that, please? :laughing: 

It took me so long to understand the difference between "x" and "set" x)

roberth commented on src/libutil/terminal.hh:
```diff
@@ -5,11 +5,43 @@
+ * more specific. Think about how the output should work with a pager or
+ * entirely non-interactive scripting use.
+ *
+ * The user may be redirecting the Lix output to a pager, but have stderr
```
```suggestion
 * The user may be redirecting the Nix output to a pager, but have stderr
```

Mic92 replied:
done.

roberth reviewed (dismissed):
Needs tests and docs

xokdvium commented:
@Mic92 Needs a rebase

Mic92 committed:
implement CLICOLOR_FORCE environment variable

This allows to force colors but not disable other features such as the
progress bar. This is for instance useful in CI environments.

Adapted from
https://git.lix.systems/lix-project/lix/commit/378ec5fb0611e314511a7afc806664205846fc2e
and others.

musjj commented:
Does this PR supersede https://github.com/NixOS/nix/pull/7650?
